### PR TITLE
Refactor shoot namespaces deployment into component

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity quality robustness
/kind technical-debt test
/priority normal
/topology seed
/platform all
/exp intermediate

**What this PR does / why we need it**:
This PR refactors the deployment for the shoot namespaces and adopts the component concept. Consequently, it eliminates the Helm chart and adds full unit tests, instead. This improves several aspects and the overall code quality.

**Which issue(s) this PR fixes**:
Part of #2754

**Special notes for your reviewer**:
Along the way I moved the `{Deploy,Delete}Namespace` functions into the new `namespaces.go` files and added some unit test for it.

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by timebertt — APPROVED** [review]
Thanks, clean refactoring!
/lgtm

File: pkg/operation/botanist/namespaces.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		return err
+	}
+
+	b.SeedNamespaceObject = namespace
+	return nil
+}
+
+// DeleteSeedNamespace deletes the namespace in the Seed cluster which holds the control plane components. The built-in
+// garbage collection in Kubernetes will automatically delete all resources which belong to this namespace. This
+// comprises volumes and load balancers as well.
```
Comment (vpnachev) [inline pkg/operation/botanist/namespaces.go:68]:
```suggestion
// comprises volumes claims and load balancers as well.
```

**Review by vpnachev — APPROVED** [review]
/lgtm

Just a small nit in the documentation comment.

File: pkg/operation/botanist/namespaces.go (hunk 1 above)
Comment (rfranzke) [reply pkg/operation/botanist/namespaces.go:68]: I'll fix it in another PR, I don't want to trigger the whole CI+approval flow now again. Thanks.

**Pull request merged**